components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java

```diff
@@ -149,10 +154,11 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         String endpointName = WebsocketEndpointManager.getInstance().getEndpointName(port, tenantDomain);
-        if (endpointName == null) {
+        if (endpointName == null && !isHealthCheckCall) {
```

Comment:
If this is a health check request, do we need to perform any of this inside this method? I think we can skip whatever we do in this method and return for a health check request. WDYT?

Reply:
Added a condition to check isHealthCheckCall at top of the method.